libsimulator/src/Ellipse.cpp

```diff
@@ -27,25 +66,35 @@ double Ellipse::EffectiveDistanceToEllipse(
     const Point& orientation_first,
     const Point& orientation_second) const
 {
-    Point R1, R2;
-    Point E1inE2, // center of E1 in coordinate system of E2
-        E2inE1;
-    E2inE1 = center_second.TransformToEllipseCoordinates(
+    Point E2inE1 = center_second.TransformToEllipseCoordinates(
         center_first, orientation_first.x, orientation_first.y);
-    E1inE2 = center_first.TransformToEllipseCoordinates(
+    Point E1inE2 = center_first.TransformToEllipseCoordinates(
         center_second, orientation_second.x, orientation_second.y);
-    // distance between centers of E1 and E2
-    const auto dist = (center_first - center_second).Norm();
-    R1 = this->PointOnEllipse(E2inE1, scale_first, center_first, speed_first, orientation_first);
-    R2 = E2.PointOnEllipse(E1inE2, scale_second, center_second, speed_second, orientation_second);
-    // effective distance
-    return dist - (center_first - R1).Norm() - (center_second - R2).Norm();
+
+    Point R1 =
+        this->PointOnEllipse(E2inE1, scale_first, center_first, speed_first, orientation_first);
+    Point R2 =
+        E2.PointOnEllipse(E1inE2, scale_second, center_second, speed_second, orientation_second);
+
+    return (R1 - R2).Norm();
 }
 
-// input: P is a point in the ellipse world.
-// output: The point on the ellipse (in cartesian coord) that lays on the same line OP
-// O being the center of the ellipse
-// if P approx equal to Center of ellipse return cartesian coordinats of the point (a,0)/ellipse
+/**
+ * @brief Computes the point on the ellipse boundary along the line from the ellipse center to point
+ * P.
+ *
+ * Given a point \( P \) in the local coordinate system of the ellipse, this function finds the
+ * corresponding point on the ellipse that lies on the same line extending from the center through
+ * \( P \).
+ *
+ * **Behavior:**
+ * - If \( P \) is very close to the ellipse center, it defaults to returning the point \((a, 0)\)
+ * on the ellipse.
+ * - Otherwise, it scales the direction from the center to \( P \) by the ellipse’s semi-major and
+ * semi-minor axes.
+ *
+ * @return Point on the ellipse boundary, transformed into the global coordinate system.
+ */
```

Comment:
Please follow the established documentation style and use /// to indicate doc-style comments. Please omit the brief because we usually use Doxygen with auto-brief, i.e. First line / first sentence is the brief.